src/shogun/kernel/DotKernel.h

```diff
@@ -68,8 +68,19 @@ class CDotKernel : public CKernel
 
 			ASSERT(l->has_property(FP_DOT))
 			ASSERT(r->has_property(FP_DOT))
-			ASSERT(l->get_feature_type() == r->get_feature_type())
-			ASSERT(l->get_feature_class() == r->get_feature_class())
+			ASSERT(l->get_feature_type() == r->get_feature_type());
+			if (l->support_compatible_class())
```

Comment:
the new Kernel class now can supports computation between difference features classes if the features in left hand side support compatible computation.